Graphics/VulkanEngine.cpp

```diff
@@ -46,27 +57,133 @@ void VulkanEngine::Initialize()
 	// Load the core Vulkan structures
 	InitializeVulkan();
 	InitializeSwapChain();
+	InitCommands();
+	InitializeDefaultRenderPass();
+	InitializeFramebuffers();
+	InitializeSyncStructures();
+	InitializePipelines();
+	LoadMeshes();
+	InitializeScene();
 
 	m_IsInitialized = true;
 	std::cout << "VulkanEngine initialized\n";
 }
 
 void VulkanEngine::Draw()
 {
-	
+	// Wait for the GPU to be done with the last frame, 1s timeout
+	VK_CHECK(vkWaitForFences(m_Device, 1, &m_RenderFence, true, 1000000000));
```

Comment:
Replace with vkGetFenceStatus
Don't hardcode timeouts

Reply:
Tested it with 10k objects with a total of 22.5mil triangles, double buffering and benchemarked it: there is no noticable improvement in frametimes.
Samsung recommends against waiting for fences on their mobile devices (and probably other mobile devices in general) since the CPU might be waiting a noticable amount of time for the GPU to finish rendering. While the CPU is waiting, it could then execute small jobs that take less than 1ms, as to not make the GPU wait whenever it is ready.
vkWaitForFences offers a lot of flexibility and is well optimised by the drivers and API. I will, however, replace the hardcoded timeout by UINT64_MAX.